bftengine/src/bcstatetransfer/SourceSelector.cpp

```diff
@@ -143,7 +149,7 @@ bool SourceSelector::shouldReplaceSource(uint64_t currTimeMilli, bool badDataFro
     return true;
   }
 
-  if (sourceReplacementTimeoutMilli_ > 0) {
+  if (lastInBatch && (sourceReplacementTimeoutMilli_ > 0)) {
```

Comment:
Should you remove current source from preferred replicas in case of timeout? Maybe updateSource() needs to handle this case differently?

Reply:
@mdarade , I think he doesn't do this, at least not at the current code I'm reviewing. see line 97. Am I missing something?